File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        def back(n):
            if n==1 or n==2:
                return se[n]
            else:
                if n not in se:
                    se[n] = back(n-1) + back(n-2)
            return se[n]
        se = {1:1 , 2:2}
        return back(n)
            

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(2 , -1);
        dp[0] = 0;
        dp[1] = 1;
        if (n==1) return 1;
        int ans;
        for (int i = 2 ; i<=n+1 ; i++){
            ans = dp[0] + dp[1];
            dp[0] = dp[1];
            dp[1] = ans;
        }
        return dp[1];
    }
};